Python 3.8 support?

Hi!

I'm one of the maintainers of the [immutables feedstock at conda-forge](https://github.com/conda-forge/immutables-feedstock). With the recent release of Python 3.8, conda-forge has started to build its packages for that new version.

Immutables builds cleanly, but one test fails on Python 3.8 (in Windows, Linux and OSX):

```
+ pytest tests
============================= test session starts ==============================
platform linux -- Python 3.8.0, pytest-3.2.2, py-1.8.0, pluggy-0.4.0
rootdir: $SRC_DIR, inifile:
collected 114 items

tests/test_map.py ...........................................................................................................F......

=================================== FAILURES ===================================
___________________________ CMapTest.test_map_pickle ___________________________
TypeError: cannot pickle 'immutables._map.MapMutation' object

During handling of the above exception, another exception occurred:

self = <tests.test_map.CMapTest testMethod=test_map_pickle>

    def test_map_pickle(self):
        h = self.Map(a=1, b=2)
        for proto in range(pickle.HIGHEST_PROTOCOL):
            p = pickle.dumps(h, proto)
            uh = pickle.loads(p)
    
            self.assertTrue(isinstance(uh, self.Map))
            self.assertEqual(h, uh)
    
        with self.assertRaisesRegex(TypeError, "can't pickle"):
>           pickle.dumps(h.mutate())
E           AssertionError: "can't pickle" does not match "cannot pickle 'immutables._map.MapMutation' object"

tests/test_map.py:1298: AssertionError
===================== 1 failed, 113 passed in 7.77 seconds =====================
```

Link: https://github.com/conda-forge/immutables-feedstock/pull/10

I will open a PR shortly.

--- a/.gitignore
+++ b/.gitignore
@@ -1,5 +1,6 @@
 *._*
 *.pyc
+*.pyd
 *.pyo
 *.scssc
 *.so

--- a/.travis.yml
+++ b/.travis.yml
@@ -3,7 +3,7 @@ language: generic
 env:
     global:
         - PYMODULE=immutables
-        - RELEASE_PYTHON_VERSIONS="3.5 3.6 3.7"
+        - RELEASE_PYTHON_VERSIONS="3.5 3.6 3.7 3.8"
 
         - S3_UPLOAD_USERNAME=oss-ci-bot
         - S3_UPLOAD_BUCKET=magicstack-oss-releases
@@ -37,6 +37,12 @@ matrix:
           env: BUILD=tests
 
         - os: linux
+          dist: xenial
+          language: python
+          python: "3.8"
+          env: BUILD=tests
+
+        - os: linux
           dist: trusty
           sudo: required
           language: python
@@ -52,6 +58,9 @@ matrix:
 
         - os: osx
           env: BUILD=tests,wheels PYTHON_VERSION=3.7.0
+
+        - os: osx
+          env: BUILD=tests,wheels PYTHON_VERSION=3.8.0
 
 cache:
     pip

--- a/setup.py
+++ b/setup.py
@@ -48,6 +48,7 @@ setuptools.setup(
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
         'Operating System :: POSIX',
         'Operating System :: MacOS :: MacOS X',
         'Operating System :: Microsoft :: Windows',

--- a/tests/test_map.py
+++ b/tests/test_map.py
@@ -1294,7 +1294,7 @@ class BaseMapTest:
             self.assertTrue(isinstance(uh, self.Map))
             self.assertEqual(h, uh)
 
-        with self.assertRaisesRegex(TypeError, "can't pickle"):
+        with self.assertRaisesRegex(TypeError, "can('t|not) pickle"):
             pickle.dumps(h.mutate())
 
     @unittest.skipIf(sys.version_info < (3, 7, 0), "__class_getitem__ is not available")

